### PR TITLE
Disc Boot Blocker + Disc Folder

### PR DESCRIPTION
**Pull request by Megamouse**
Adds dev_hdd0/disc for disc games.

### CODE REVIEW

File: rpcs3/rpcs3qt/game_list_frame.cpp [hunk 1]
```diff
@@ -500,11 +510,19 @@ void game_list_frame::doubleClickedSlot(const QModelIndex& index)
...
-	
+	std::string game_dir	= m_game_data[i].dir;
+
 	// Boot these categories
 	if (category == category::hdd_Game || category == category::disc_Game || category == category::audio_Video)
 	{
-		const std::string& path = Emu.GetGameDir() + m_game_data[i].info.root;
+		const std::string& path = game_dir + m_game_data[i].info.root;
+
+		if (qstr(path).contains("/dev_hdd0/game/") && fs::is_file(path + "/PS3_GAME/USRDIR/EBOOT.BIN") && fs::is_file(path + "/PS3_DISC.SFB"))
```
Comment (AniLeo) [inline rpcs3/rpcs3qt/game_list_frame.cpp]: Some people delete SFB files or have them renamed. Better to make it OR instead of AND

**Comment by Megamouse** [comment]
@AniLeo I used tolower in newest commit

File: rpcs3/Emu/System.cpp [hunk 2]
```diff
@@ -218,10 +219,22 @@ void Emulator::Init()
...
 }
 
-void Emulator::SetPath(const std::string& path, const std::string& elf_path)
+bool Emulator::SetPath(const std::string& path, const std::string& elf_path)
 {
+	std::string str = path;
+	std::transform(str.begin(), str.end(), str.begin(), ::tolower);
+
+	if (str.find("/dev_hdd0/game/") != std::string::npos && str.find("/ps3_game/usrdir/") != std::string::npos ||
+		str.find("\\dev_hdd0\\game\\") != std::string::npos && str.find("\\ps3_game\\usrdir\\") != std::string::npos)
```
Comment (Nekotekina) [inline rpcs3/Emu/System.cpp]: This part doesn't really make sense this way...

**Comment by Nekotekina** [comment]
Moving a folder would be a one-line code... And if we can do it, we don't need to block it.

```C++
fs::rename(x, y);
```

The way to detect disc games... varies a bit. We load PARAM.SFO in System.cpp.

**Comment by Megamouse** [comment]
so when should that move be executed?
on rpcs3 boot or on booting of that game?

File: rpcs3/rpcs3qt/game_list_frame.cpp [hunk 3]
```diff
@@ -352,28 +352,47 @@ void game_list_frame::Refresh(bool fromDrive)
 		const std::string& game_path = Emu.GetGameDir();
 		const std::string& disc_path = Emu.GetDiscDir();
 
-		for (auto path : { game_path, disc_path })
+		for (const auto& path : { game_path, disc_path })
 		{
 			for (const auto& entry : fs::dir(path))
 			{
-				if (!entry.is_directory)
+				if (!entry.is_directory || entry.name == "." || entry.name == ".." | entry.name == "TEST12345")
```
Comment (Farseer2) [inline rpcs3/rpcs3qt/game_list_frame.cpp]: Don't you mean || ?

Comment (Megamouse) [inline rpcs3/rpcs3qt/game_list_frame.cpp]: oh crap. thx. not that it really matters xD

**Pull request closed without merging**